api/src/services/uploads/uploads.ts

```diff
@@ -258,3 +258,42 @@ export const getUploadsByExpenditureCategory = async () => {
 
   return uploadsByExpenditureCategory
 }
+
+export const sendTreasuryReport: MutationResolvers['sendTreasuryReport'] =
+  async () => {
+    try {
+      const organization = await db.organization.findFirst({
+        where: { id: context.currentUser.agency.organizationId },
+      })
+      const reportingPeriod = await db.reportingPeriod.findFirst({
+        where: { id: organization.preferences['current_reporting_period_id'] },
+      })
+      const uploadsByExpenditureCategory =
+        await getUploadsByExpenditureCategory()
+      const arn = process.env.TREASURY_STEP_FUNCTION_ARN
+
+      if (!arn) {
+        throw new Error('TREASURY_STEP_FUNCTION_ARN is not set')
+      }
+      logger.info(uploadsByExpenditureCategory)
+      logger.info('Sending Treasury Report')
+
+      const input = {
+        reportingPeriod: reportingPeriod.name,
+        organization: organization.name,
+        email: context.currentUser.email,
+        uploadsByExpenditureCategory,
+      }
+
+      await aws.startStepFunctionExecution(
+        arn,
+        undefined,
```

Comment:
As per [docs](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/client/sfn/command/StartExecutionCommand/): "Optional name of the execution. This name must be unique for your Amazon Web Services account, Region, and state machine for 90 days. For more information, see [Limits Related to State Machine Executions ](https://docs.aws.amazon.com/step-functions/latest/dg/limits.html#service-limits-state-machine-executions)
 in the Step Functions Developer Guide.

If you don't provide a name for the execution, Step Functions automatically generates a universally unique identifier (UUID) as the execution name."

Let me know if we would prefer to name this execution explicitly.

Reply:
I think its fine to leave this alone for now and use the UUID default. 👍